src/server.lisp

```diff
@@ -147,123 +147,132 @@ By default, a symbol passed in for F will be automatically converted into the na
   |error|)
 
 
+(defun log-completion-message (logger request reply start-time)
+  (macrolet ((log-it (priority control &rest args)
+               `(cl-syslog:rfc-log
+                    (logger ,priority ,control ,@args)
+                  (:msgid "LOG0002")
+                  (|rigetti@0000|
+                   |methodName| (|RPCRequest-method| request)
+                   |requestID| (|RPCRequest-id| request)
+                   |wallTime| (format nil "~f" (/ (- (get-internal-real-time) start-time)
+                                                  internal-time-units-per-second))
+                   |error| ,(if (<= (cl-syslog:get-priority priority)
+                                    (cl-syslog:get-priority ':err))
+                                "true"
+                                "false")))))
+    (etypecase reply
+      (|RPCReply|
+       (log-it :info "Requested ~a completed" (|RPCRequest-method| request)))
+      (|RPCError|
+       (log-it :err "Request ~a error: ~a" (|RPCRequest-id| request) (|RPCError-error| reply))))))
+
+
+(defvar *warnings*)
+(setf (documentation '*warnings* 'variable)
+      "An adjustable array of |RPCWarning|s that correspond to WARNINGs signaled during the dynamic extent of %PROCESS-RAW-REQUEST.
+
+These warnings are included in the RPC response that is returned to the caller.
+
+*WARNINGS* is unbound in the global environment.")
+
+(defun %process-request (request dispatch-table timeout debug)
+  "Process the given |RPCRequest| REQUEST and return either an |RPCReply| or an |RPCError|."
+  (flet ((error-processor (c h)
+           (declare (ignore h))
+           ;; this is where error handlers go for errors where we can reply to the client
+           (return-from %process-request
+             (typecase c
+               (unknown-rpc-method
+                (make-instance '|RPCError|
+                               :|id| (|RPCRequest-id| request)
+                               :|error| (format nil "Method named \"~a\" is unknown."
+                                                (|RPCRequest-method| request))
+                               :|warnings| *warnings*))
+               (bt:timeout
+                (make-instance '|RPCError|
+                               :|id| (|RPCRequest-id| request)
+                               :|error| (format nil "Execution timed out.  Note: time limit: ~a seconds." timeout)
+                               :|warnings| *warnings*))
+               (otherwise
+                (make-instance '|RPCError|
+                               :|id| (|RPCRequest-id| request)
+                               :|error| (format nil "Unhandled error in host program:~%~a" c)
```

Comment:
N.B. error message has changed here slightly to avoid the need to special-case logging it.